outputs/otlp.go

```diff
@@ -0,0 +1,172 @@
+	if err != nil {
+		return nil, err
+	}
+	shutDownFunc, err := otlpInit(config)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("[INFO] : OTLP.Traces=%+v\n", config.OTLP.Traces)
+	otlpClient.ShutDownFunc = shutDownFunc
+	return otlpClient, nil
+}
+
+// newTrace returns a new Trace object.
+func (c *Client) newTrace(falcopayload types.FalcoPayload) *trace.Span {
+	traceID, _, err := generateTraceID(falcopayload, c.Config)
+	if err != nil {
+		log.Printf("[ERROR] : OLTP Traces - Error generating trace id: %v for output fields %v", err, falcopayload.OutputFields)
+		return nil
+	}
+
+	startTime := falcopayload.Time
+	endTime := falcopayload.Time.Add(time.Millisecond * time.Duration(c.Config.OTLP.Traces.Duration))
+
+	sc := trace.SpanContext{}.WithTraceID(traceID)
+	ctx := trace.ContextWithSpanContext(context.Background(), sc)
+
+	tracer := getTracerProvider().Tracer("falco-event")
+	_, span := tracer.Start(
+		ctx,
+		falcopayload.Rule,
+		trace.WithTimestamp(startTime),
+		trace.WithSpanKind(trace.SpanKindServer))
+
+	span.SetAttributes(attribute.String("uuid", falcopayload.UUID))
+	span.SetAttributes(attribute.String("source", falcopayload.Source))
+	span.SetAttributes(attribute.String("priority", falcopayload.Priority.String()))
+	span.SetAttributes(attribute.String("rule", falcopayload.Rule))
+	span.SetAttributes(attribute.String("output", falcopayload.Output))
+	span.SetAttributes(attribute.String("hostname", falcopayload.Hostname))
+	span.SetAttributes(attribute.StringSlice("tags", falcopayload.Tags))
+	for k, v := range falcopayload.OutputFields {
+		span.SetAttributes(attribute.String(k, fmt.Sprintf("%v", v)))
+	}
+	// span.AddEvent("falco-event")
+	span.End(trace.WithTimestamp(endTime))
+
+	if c.Config.Debug {
+		log.Printf("[DEBUG] : OTLP Traces - payload generated successfully for traceid=%s", span.SpanContext().TraceID())
+	}
+
+	return &span
+}
+
+// OTLPPost generates an OTLP trace _implicitly_ via newTrace() by
+// calling OTEL SDK's tracer.Start() --> span.End(), i.e. no need to explicitly
+// do a HTTP POST
+func (c *Client) OTLPTracesPost(falcopayload types.FalcoPayload) {
+	c.Stats.OTLP.Add(Total, 1)
+
+	trace := c.newTrace(falcopayload)
+	if trace == nil {
+		go c.CountMetric(Outputs, 1, []string{"output:otlp/traces", "status:error"})
+		c.Stats.OTLP.Add(Error, 1)
+		c.PromStats.Outputs.With(map[string]string{"destination": "otlp/traces", "status": Error}).Inc()
+		log.Printf("[ERROR] : OLTP Traces - Error generating trace")
+		return
+	}
+	// Setting the success status
+	go c.CountMetric(Outputs, 1, []string{"output:otlp/traces", "status:ok"})
+	c.Stats.OTLP.Add(OK, 1)
+	c.PromStats.Outputs.With(map[string]string{"destination": "otlp/traces", "status": OK}).Inc()
+}
+
+const (
+	templateOption       = "missingkey=zero"
+	kubeTemplateStr      = `{{.cluster}}{{.k8s_pod_name}}{{.k8s_ns_name}}{{.k8s_container_name}}`
+	containerTemplateStr = `{{.container_id}}`
```

Comment:
where do you get the `.cluster` from? btw, you only address syscalls events, in a pod/container, but we need to address other use cases, like the events from the plugins, without the fields you use.

Reply:
well ... under the umbrella of «opinionated» :P ... srsly speaking, using a rather _common_ `cluster` label, but I see your point, I'm fine removing it and keeping `{{container.id}}` only as default (still letting ppl override it via `config.OTLP.Traces.TradeIDHash`).

Reply:
even if people are able to override the template, we still have the issue that the available fields are not the same between the sources